sycl/include/CL/sycl/detail/platform_impl.hpp

```diff
@@ -8,8 +8,9 @@
 
 #pragma once
 #include <CL/sycl/detail/common.hpp>
-#include <CL/sycl/detail/platform_info.hpp>
+#include <CL/sycl/detail/force_device.hpp>
```

Comment:
I moved them into "detail" headers, which seemed to be consistent with the overall structure. If you inisist, I can copy back to the sources, but prefer doing so in a separate subsequent change-set. Please let me know.

Reply:
I think the right place for this header is source directory because it doesn't declare **public** interface, so SYCL library users are not supposed to include this header fie. 
It must be included into `sycl/source/*` files only.

Reply:
Nothing in the "sycl/detail" is supposed to be included directly by SYCL users, and this one is not an exception to this. If this was really the goal (not to let people see the detail) we could have gone too much further in separation, but is is the goal?

Reply:
The reason why we have "sycl/detail" sub-directory in public interface is because they contain implementation of template classes/methods, which can't be moved to the source directory, but it's not the case with `force_device.hpp`.

Reply:
No problem, we can "hide" this one and many more from sycl/detail if we wanted to.